src/commands/deploy/deploy.ts

```diff
@@ -834,6 +853,12 @@ export const deploy = async (options: OptionValues, command: BaseCommand) => {
     }
   }
 
+  if (!siteId) {
+    return logAndThrowError(
+      "Unable to determine which site to deploy to. Make sure you've run 'netlify link' or that you're specifying your desired site using the '--site' option.",
+    )
+  }
+
```

Comment:
I think you only need this because `initChoice` is untyped above, so TS doesn't know that the checks above are exhaustive?

I believe you can just add `as const` to line 834 and `as {initChoice: typepof initializeOpts[number]}` on line 842. Come to think of it, I think I have this on a branch somewhere...

Reply:
This doesn't have anything to do with `initChoice`--need this because the `if`/`else if` conditions on L837-845 are not exhaustive and do not unconditionally set `siteId`.

Reply:
No, the conditions are exhaustive. You need this because `initChoice` is `any` and `siteId` is `any`. I'll open a PR.

Reply:
https://github.com/netlify/cli/pull/7220